Bug: gcu: generic config updater test_pfcwd_status.py test failure

### Is it platform specific

generic

### Importance or Severity

Medium

### Description of the bug

generic_config_updater/test_pfcwd_status.py failure
Test failure is observed on marvell-teralynx platform.

### Steps to Reproduce

Run generic_config_updater/test_pfcwd_status.py test case
```
./run_tests.sh -n ptf2-m -d str-marvell-01 -c "generic_config_updater/test_pfcwd_status.py::test_start_pfcwd[single]" -f ../ansible/testbed.yaml -i ../ansible/lab,../ansible/veos -t t0,any -u
```

### Actual Behavior and Expected Behavior

Actual behavior is test case failure.
```
generic_config_updater/test_pfcwd_status.py::test_start_pfcwd[single] FAILED             [ 75%]  
generic_config_updater/test_pfcwd_status.py::test_start_pfcwd[all] FAILED                   [100%]                                                                                                 
```

Test case is expected to pass

### Relevant log output

```shell
output = {'changed': True, 'stdout': 'Patch Applier: localhost: Patch application starting.\nPatch Applier: localhost: Patch: [... application completed.', 'Patch applied successfully.'], 'stderr_lines': [], '_ansible_no_log': None, 'failed': False}

    def expect_op_failure(output):
        """Expected failure from apply-patch output
        """
        logger.info("Return code: {}, error: {}".format(output['rc'], output['stderr']))
>       pytest_assert(
            output['rc'],
            "The command should fail with non zero return code"
        )
E       Failed: The command should fail with non zero return code
```

### Output of `show version`

```shell
SONiC Software Version: SONiC.202411.31-dirty-20250428.114028
SONiC OS Version: 12
Distribution: Debian 12.9
Kernel: 6.1.0-29-2-amd64
Build commit: e297e6f93
Build date: Mon Apr 28 12:44:03 UTC 2025
Built by: marvell@cpss-testvm2

Platform: x86_64-marvell_dbmvtx9180-r0
HwSKU: dbmvtx9180_64osfp_128x50G
ASIC: marvell-teralynx
ASIC Count: 1
Serial Number: N/A
Model Number: N/A
Hardware Revision: N/A
Uptime: 09:13:55 up  2:56,  1 user,  load average: 0.63, 1.30, 1.51
Date: Fri 06 Dec 2024 09:13:55

Docker images:
REPOSITORY                       TAG                               IMAGE ID       SIZE
docker-macsec                    latest                            b826b0310105   344MB
docker-dhcp-relay                latest                            17f9d511beb4   321MB
docker-syncd-mrvl-teralynx-rpc   202411.31-dirty-20250428.114028   12c57591e69d   635MB
docker-sonic-telemetry           202411.31-dirty-20250428.114028   759a91e72425   402MB
docker-sonic-telemetry           latest                            759a91e72425   402MB
docker-teamd                     202411.31-dirty-20250428.114028   e6c2b1018e11   341MB
docker-teamd                     latest                            e6c2b1018e11   341MB
docker-sonic-mgmt-framework      202411.31-dirty-20250428.114028   94f2c68516cc   400MB
docker-sonic-mgmt-framework      latest                            94f2c68516cc   400MB
docker-snmp                      202411.31-dirty-20250428.114028   1e82a07fad54   356MB
docker-snmp                      latest                            1e82a07fad54   356MB
docker-sflow                     202411.31-dirty-20250428.114028   2318dcde1f24   342MB
docker-sflow                     latest                            2318dcde1f24   342MB
docker-router-advertiser         202411.31-dirty-20250428.114028   1cf1179e0f0b   312MB
docker-router-advertiser         latest                            1cf1179e0f0b   312MB
docker-platform-monitor          202411.31-dirty-20250428.114028   31801fefe3a4   432MB
docker-platform-monitor          latest                            31801fefe3a4   432MB
docker-orchagent                 202411.31-dirty-20250428.114028   a9a554ba179a   354MB
docker-orchagent                 latest                            a9a554ba179a   354MB
docker-nat                       202411.31-dirty-20250428.114028   220a800d4c7b   344MB
docker-nat                       latest                            220a800d4c7b   344MB
docker-mux                       202411.31-dirty-20250428.114028   c2acee6182a2   364MB
docker-mux                       latest                            c2acee6182a2   364MB
docker-lldp                      202411.31-dirty-20250428.114028   b68d81e84f00   358MB
docker-lldp                      latest                            b68d81e84f00   358MB
docker-iccpd                     202411.31-dirty-20250428.114028   3b692408c925   344MB
docker-iccpd                     latest                            3b692408c925   344MB
docker-sonic-gnmi                202411.31-dirty-20250428.114028   ba0d36dff0d5   402MB
docker-sonic-gnmi                latest                            ba0d36dff0d5   402MB
docker-fpm-frr                   202411.31-dirty-20250428.114028   bf9cc4240a0d   375MB
docker-fpm-frr                   latest                            bf9cc4240a0d   375MB
docker-eventd                    202411.31-dirty-20250428.114028   9feaa5ab39d5   312MB
docker-eventd                    latest                            9feaa5ab39d5   312MB
docker-database                  202411.31-dirty-20250428.114028   be38cf3fe15a   320MB
docker-database                  latest                            be38cf3fe15a   320MB
docker-sonic-bmp                 202411.31-dirty-20250428.114028   bbf39434c75d   313MB
docker-sonic-bmp                 latest                            bbf39434c75d   313MB
docker-syncd-mrvl-teralynx-rpc   latest                            245877040743   845MB
```

### Attach files (if any)

[pfcwd_status_failure_test.log](https://github.com/user-attachments/files/20241949/pfcwd_status_failure_test.log)

--- a/tests/common/devices/sonic.py
+++ b/tests/common/devices/sonic.py
@@ -1862,6 +1862,8 @@ Totals               6450                 6449
             asic = "gb"
         elif "Mellanox Technologies" in output:
             asic = "spc"
+        elif "Marvell Technology" in output:
+            asic = "marvell-teralynx"
 
         logger.info("asic: {}".format(asic))
 

--- a/tests/common/gu_utils.py
+++ b/tests/common/gu_utils.py
@@ -454,8 +454,8 @@ def is_valid_platform_and_version(duthost, table, scenario, operation, field_val
             "validator_data"]["rdma_config_update_validator"][scenario]["platforms"][asic]     # noqa: E501
         if version_required == "":
             return False
-        # os_version is in format "20220531.04", version_required is in format "20220500"
-        return os_version[0:8] >= version_required[0:8]
+        # os_version can be in any of 202411.1, 20241100.1 formats and version_required is in format "20220500"
+        return os_version.split('.')[0].ljust(8, '0') >= version_required[0:8]
     except KeyError:
         return False
     except IndexError:

